src/main/resources/static/assets/js/sdh.js

```diff
@@ -73,8 +75,84 @@
 		    _isShowDetails(field) {
 		    	return field.key === 'show_details';
 		    },
+		    _isEdit(field) {
+		    	return field.key === 'edit';
+		    },
 		    _formatDate(dateString) {
 		    	return dateString.replace(/ \d\d:\d\d:\d\d\.\d/, '');
+		    },
+		    edit(item, index, target) {
+		    	this.modalEditRow.domain = this.domainOptions.filter(it => it.description === item.domain)[0].id;
+		    	if (item.id === "") {
+		    		this.modalEditRow.title = 'New Clinical Review';
+		    	} else {
+		    		this.modalEditRow.title = 'Edit Clinical Review';
+		    		this.modalEditRow.id = item.id;
+		    	      this.modalEditRow.prioritySelected = this.modalEditRow.priorityOptions.filter(it => it.text === item.clinician_priority)[0].value;
+			        this.modalEditRow.readinessSelected = this.modalEditRow.readinessOptions.filter(it => it.text === item.patient_readiness)[0].value;
+		    	}
+				this.$root.$emit('bv::show::modal', 'modalEditRow', target);
+		    },
+		    resetModal() {
+		        this.modalEditRow.title = '';
+		        this.modalEditRow.id = '';
+		        this.modalEditRow.domain = '';
+		        this.modalEditRow.prioritySelected = '';
+		        this.modalEditRow.readinessSelected = '';
+		    },
+		    handleOk(evt) {
+		        // Prevent modal from closing so we can validate
+		        evt.preventDefault();
+		        if (!this.modalEditRow.prioritySelected || !this.modalEditRow.readinessSelected) {
+		          alert('Please enter values');
+		        } else {
+		          this.handleSubmit();
+		        }
+		    },
+		    handleSubmit() {
+				const patient = this.$el.getAttribute('data-patient-id');
+				const contextPath = this.$el.getAttribute('data-context-path');
+				let clinicianReviewId = this.modalEditRow.id;
+				let {url, method, data} = '';
+				if (clinicianReviewId) {
+					url = contextPath + '/api/clinician_review/' + clinicianReviewId;
+					method = 'PATCH';
+					data = JSON.stringify({
+						clinicianPriority: contextPath + '/api/clinician_priority/' + this.modalEditRow.prioritySelected,
+						patientReadiness: contextPath + '/api/patient_readiness/' + this.modalEditRow.readinessSelected
+					});
+				} else {
+					url = contextPath + '/api/clinician_review/';
+					method = 'POST';
+					data = JSON.stringify({
+						patient: contextPath + '/api/patient/' + patient,
+						domain: contextPath + '/api/sdh_domain/' + this.modalEditRow.domain,
+						clinicianPriority: contextPath + '/api/clinician_priority/' + this.modalEditRow.prioritySelected,
+						patientReadiness: contextPath + '/api/patient_readiness/' + this.modalEditRow.readinessSelected
+					});					
+				}
+				
+				$.ajax({
+					url: url,
+					method: method,
+					data: data,
+					contentType: 'application/json',
+					success: data => {
+						// TODO: Better way to reload without getting all reviews again
```

Comment:
I'm not quite sure what you're getting at with the TODO here, but could you potentially change the state you already have in the browser, rather than fetching it again from the server?

Reply:
Yes, that is what I initially did, but it was ugly. The list of reviews retrieved by calling "search/findByPatientId?id=" uses projections to format the reviews variable more naturally. But when you do a single retrieval, projections aren't used and the format doesn't nicely align. I'm going to try an alternative approach once I get other comments addressed.

Reply:
Sorry, I wasn't clear. I meant to suggest that you use the "optimistic UI" pattern. If you have enough information, you can fake a response from the server that has the shape that you expect, then use that to update the UI. When you finally get a response from the server, you can patch the state as needed (or not, depending on your circumstances).

Reply:
I played more with this, but it ended up being more code to update just the row that changed, partly because a row doesn't map one to one with a domain object. And when creating a new review the response doesn't return the new id - it has to be parsed from the href. For the demo, I think it's sufficient just to refresh the rows - there won't ever be more than 6 or 7.